docs/features/nullable-reference-types.md

```diff
@@ -0,0 +1,119 @@
+Nullable Reference Types
+=========
+References to reference types may be nullable, non-nullable, or null-oblivious. (The document uses the shorthand `?`, `!`, and `~`.)
+
+## Annotations
+In source, nullable reference types are annotated with `?`.
+```c#
+string? OptString; // may be null
+List<object?>? OptListOptElements; // list may be null, elements may be null
+```
+In metadata, nullable reference types are annotated with a `[Nullable]` attribute.
+```c#
+namespace System.Runtime.CompilerServices
+{
+    public sealed class NullableAttribute : Attribute
+    {
+        public NullableAttribute() { }
+        public NullableAttribute(bool[] b) { }
+    }
+}
+```
+If annotations are enabled in a module, a `[module: Nullable]` attribute is included.
+```c#
+// C# representation of metadata
+[module: Nullable]
+[Nullable] string OptString; // string?
+[Nullable(new[] { true, true })] List<object> OptListOptElements; // List<object?>?
+```
+`NullableAttribute` cannot be used explicitly in source so the type declaration can be synthesized by the compiler if it is not included in the compilation.
+
+Unannotated reference types are non-nullable or null-oblivious depending whether the containing scope includes `[NonNullTypes]`.
+```c#
+namespace System.Runtime.CompilerServices
+{
+    public sealed class NonNullTypesAttribute : Attribute
+    {
+        public NonNullTypesAttribute(bool enabled = true) { }
+    }
+}
+```
+If there is no `[NonNullTypes]` attribute in any containing scope, including the module, reference types are non-nullable.
+```c#
+[NonNullTypes(false), Nullable(new[] { false, true })] string[] Oblivious; // string?[]~
+[NonNullTypes(true), Nullable(new[] { false, true })] string[] NotNull; // string?[]!
```

Comment:
Please add one more example, with `[NonNullTypes(true)]` and without `[Nullable(...)]`.